tools/im2rec.py

```diff
@@ -1,238 +1,265 @@
+                        help='Ratio of images to use for training.')
+    cgroup.add_argument('--test-ratio', type=float, default=0,
+                        help='Ratio of images to use for testing.')
+    cgroup.add_argument('--recursive', type=bool, default=False,
+                        help='If true recursively walk through subdirs and assign an unique label\
+        to images in each folder. Otherwise only include images in the root folder\
+        and give them label 0.')
+
+    rgroup = parser.add_argument_group('Options for creating database')
+    rgroup.add_argument('--resize', type=int, default=0,
+                        help='resize the shorter edge of image to the newsize, original images will\
+        be packed by default.')
+    rgroup.add_argument('--center-crop', type=bool, default=False,
+                        help='specify whether to crop the center image to make it rectangular.')
+    rgroup.add_argument('--quality', type=int, default=80,
+                        help='JPEG quality for encoding, 1-100; or PNG compression for encoding, 1-9')
+    rgroup.add_argument('--num-thread', type=int, default=1,
+                        help='number of thread to use for encoding. order of images will be different\
+        from the input list if >1. the input list will be modified to match the\
+        resulting order.')
+    rgroup.add_argument('--color', type=int, default=1, choices=[-1, 0, 1],
+                        help='specify the color mode of the loaded image.\
+        1: Loads a color image. Any transparency of image will be neglected. It is the default flag.\
+        0: Loads image in grayscale mode.\
+        -1:Loads image as such including alpha channel.')
+    rgroup.add_argument('--encoding', type=str, default='.jpg', choices=['.jpg', '.png'],
+                        help='specify the encoding of the images.')
+    rgroup.add_argument('--saving-folder', type=str, default='.',
+                        help='folder in which .rec files will be saved.')
+    rgroup.add_argument('--shuffle', default=True, help='If this is set as True, \
+        im2rec will randomize the image order in <prefix>.lst')
+    args = parser.parse_args()
+
+    # add data_path here
+    # data_path = os.path.abspath(args.root)
+    data_path = args.root
+    if args.list:
+        make_list(args)
+    else:
+        # f is just file name original, not a path, here changed it
+        files = [f for f in os.listdir(data_path) if os.path.isfile(os.path.join(data_path, f))]
+        for f in files:
+            if f.startswith(args.prefix) is True and f.endswith('.lst') is True:
+                print 'Creating .rec file from', f, 'in', args.saving_folder
+                # join f with data_path
+                image_list = read_list(os.path.join(data_path, f))
+                # delete write record and moved it to the __main__ process
+                return (args, image_list, f)
+
+
+if __name__ == '__main__':

```

Comment:
here is the __main__ process, which do the write operation original in write_record